gst-libs/mfx/gstmfxprimebufferproxy.c

```diff
@@ -199,15 +220,15 @@ gst_mfx_prime_buffer_proxy_get_handle (GstMfxPrimeBufferProxy * proxy)
 {
   g_return_val_if_fail (proxy != NULL, 0);
 
-  return proxy->fd;
+  return proxy->buf_info.handle;
```

Comment:
proxy->fd is already assigned, so this change is not required.

Reply:
I think it makes more sense for fd to be assigned to the buf_info structure instead of the proxy structure. proxy size/handle implies the size and handle of the buffer proxy itself not the actual buffer that the buffer proxy is suppose to carry. This to me makes more logical sense.

Reply:
The reasons for the current buffer proxy implementation for getting the fd is mainly due to legacy compatibility with MSS 2016, since at the iHD driver for MSS 2016 didn't support vaAcquireBufferHandle / vaReleaseBufferHandle methods in the custom libva API that time (partly explained https://github.com/ishmael1985/gstreamer-media-SDK/commit/bcf058a15368a60fd5e364cc288c899662a5895f#diff-5d3ef4633360d029e0bc18542d4ac974). The proposed change to fix the getter to the VABufferInfo member structure  will break legacy compatibility for MSS 2016 Linux, and there's also a possibility that libva 2.0 will propose new changes to get the fd handle / size (vaExportSurfaceHandle), so I would recommend leaving it as a proxy implementation here.